chia/wallet/wallet_node.py

```diff
@@ -824,13 +824,27 @@ async def state_update_received(self, request: wallet_protocol.CoinStateUpdate,
                 request.peak_hash,
             )
 
-    def get_full_node_peer(self) -> Optional[WSChiaConnection]:
+    def get_full_node_peer(self, synced_only: bool = False) -> Optional[WSChiaConnection]:
+        """
+        Get a full node, preferring synced & trusted > synced & untrusted > unsynced & trusted > unsynced & untrusted
+        """
         if self.server is None:
             return None
 
         nodes = self.server.get_full_node_connections()
         if len(nodes) > 0:
-            return random.choice(nodes)
+            synced_peers = set(node for node in nodes if node.peer_node_id in self.synced_peers)
+            trusted_peers = set(node for node in nodes if self.is_trusted(node))
+            if len(synced_peers & trusted_peers) > 0:
+                return random.choice(list(synced_peers & trusted_peers))
+            elif len(synced_peers) > 0:
+                return random.choice(list(synced_peers))
+            elif synced_only:
```

Comment:
why would a wallet node ever need a not synced full node peer?

Reply:
Some of the calls may succeed because the nodes are synced _enough_.